src/openqaoa-core/tests/test_workflows.py

```diff
@@ -1726,6 +1731,640 @@ def test_change_properties_after_compilation(self):
+            ):
+                fqaoa.evaluate_circuit(params2 + [1])
+
+            # evaluate the circuit with a list shorter than it should, it should raise an error
+            with self.assertRaises(
+                AssertionError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                passing a list shorter than it should"
+            ):
+                fqaoa.evaluate_circuit(params2[:-1])
+
+            # evaluate the circuit with a dict with a wrong key, it should raise an error
+            with self.assertRaises(
+                KeyError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error \
+                when passing a dict with a wrong key"
+            ):
+                fqaoa.evaluate_circuit({**params, "wrong_key": 1})
+
+            # evaluate the circuit with a dict with a value longer than it should, it should raise an error
+            with self.assertRaises(
+                ValueError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                passing a dict with a value longer than it should"
+                ):
+                fqaoa.evaluate_circuit(
+                    {**params, list(params.keys())[0]: np.random.rand(40)}
+                )
+
+            # evaluate the circuit without passing any param, it should raise an error
+            with self.assertRaises(
+                TypeError,
+                msg=f"param_type={fqaoa.circuit_properties.param_type}. `evaluate_circuit` should raise an error when \
+                    not passing any param",
+                ):
+                fqaoa.evaluate_circuit()
+
+    def test_qaoa_evaluate_circuit_shot(self):
+        # problem
+        num_assets, budget = 5, 3
+        mu, sigma, _ = generate_random_portfolio_data(num_assets = num_assets, num_days = 15, seed = 1)
+        problem = PortfolioOptimization(mu, sigma, risk_factor = None, budget = budget, penalty = None).qubo
+
+        if "qiskit.qasm_simulator" not in SUPPORTED_LOCAL_SIMULATORS:
+            self.skipTest(
+                reason="Qiskit QASM Simulator is not available. Please install the qiskit plugin: openqaoa-qiskit."
+            )
+        else:
+            # check that it works with shots
+            fqaoa = FQAOA()
+            device = create_device(location="local", name="qiskit.qasm_simulator")
+            fqaoa.set_device(device)
+            fqaoa.set_circuit_properties(p=3)
+
+            # try to evaluate the circuit before compiling
+            error = False
+            try:
+                fqaoa.evaluate_circuit()
+            except Exception:
+                error = True
+            assert (
+                error
+            ), f"param_type={param_type}. `evaluate_circuit` should raise an error if the circuit is not compiled"
```

Comment:
I think this code is outdated, same in `TestingVanillaQAOA.test_evaluate_circuit_shot`, let's use the error context manager here instead:
```python
with self.assertRaises(ValueError) as cm:
  fqaoa.evaluate([1, 2, 1, 2, 1, 2])
self.assertIn(
            "Please compile the QAOA before optimizing it!",
            str(cm.exception),
        )
```
Note that `evaluate` requires to input the parameter value. We also test for the error message to make sure we raise not only the right type of error but the exact error.

Can you please also replace this in `TestingVanillaQAOA.test_evaluate_circuit_shot`?

Reply:
Use error context manager here and 
replace this in `TestingVanillaQAOA.test_evaluate_circuit_shot`.